ldhooks/evaluation_series_data.go

```diff
@@ -0,0 +1,77 @@
+package ldhooks
+
+// EvaluationSeriesData is an immutable data type used for passing implementation-specific data between stages in the
+// evaluation series.
+type EvaluationSeriesData struct {
+	data map[string]any
+}
+
+// EvaluationSeriesDataBuilder should be used by hook implementers to append data
+type EvaluationSeriesDataBuilder struct {
+	data map[string]any
+}
+
+// EmptyEvaluationSeriesData returns empty series data. This function is not intended for use by hook implementors.
+// Hook implementations should always use NewEvaluationSeriesBuilder.
+func EmptyEvaluationSeriesData() EvaluationSeriesData {
+	return EvaluationSeriesData{
+		data: make(map[string]any),
+	}
+}
+
+// Get gets the value associated with the given key. If there is no value, then ok will be false.
+func (b EvaluationSeriesData) Get(key string) (value any, ok bool) {
+	val, ok := b.data[key]
+	return val, ok
+}
+
+// AsAnyMap returns a copy of the contents of the series data as a map.
+func (b EvaluationSeriesData) AsAnyMap() map[string]any {
```

Comment:
https://pkg.go.dev/maps#Clone

Reply:
I think this isn't an immutable map. Since the caller could call `AsAnyMap()` and then modify the values, which are shallow cloned. 

E.g.
```golang
foo := data.AsAnyMap()
foo["bar"].someField = 3

assert.Equal(data["bar"].someField, 3)
```

maybe?

Reply:
Unless I am misunderstanding. The map itself is immutable. You cannot add new keys to it and you cannot replace values in it.

The values themselves may or may not be mutable depending on their type. Which is how I anticipate our implementations being in any language. (Which is also the purpose, calling `span.end()` for instance on a span we get out of the map.

You cannot change the original input map itself though (add keys to it, assign different values to keys).

Reply:
No, you're right, I was under the temporary illusion that I had (or you had specified) that the value was an `LdValue`.